FlowCrypt/Controllers/Msg/MessageViewController.swift

```diff
@@ -144,35 +147,36 @@ final class MessageViewController: TableNodeViewController {
 extension MessageViewController {
     private func fetchDecryptAndRenderMsg() {
         showSpinner("loading_title".localized, isUserInteractionEnabled: true)
-
-        Promise { [weak self] in
-            guard let self = self else { return }
-            let promise = self.messageService.getAndProcessMessage(
-                with: self.input.objMessage,
-                folder: self.input.path
-            )
-            let message = try awaitPromise(promise)
-            self.processedMessage = message
-        }
-        .then(on: .main) { [weak self] in
-            self?.handleReceivedMessage()
-        }
-        .catch(on: .main) { [weak self] error in
-            self?.handleError(error)
+        Task {
+            do {
+                let rawMimeData = try awaitPromise(self.messageProvider.fetchMsg(message: self.input.objMessage, folder: self.input.path))
+                self.processedMessage = try await self.messageService.decryptAndProcessMessage(mime: rawMimeData)
+                DispatchQueue.main.async {
```

Comment:
You should be on main thread here

Reply:
Hm. Exactly the following two should not run on main thread:

```
                let rawMimeData = try awaitPromise(self.messageProvider.fetchMsg(message: self.input.objMessage, folder: self.input.path))
```

How to change it to make sure these two run on other thread?

Reply:
You could use actors like I did for other controllers until we move service layer to actors completely